app/controllers/application_controller.rb

```diff
@@ -608,12 +620,27 @@ def should_check_authorization?
   end
 
   def validate_headers
-    if response.headers.has_key?('Content-Length') && response.headers['Content-Length'].to_i < 0
+    if response.headers.key?('Content-Length') && response.headers['Content-Length'].to_i.negative?
       raise CustomErrors::BadHeaderError
     end
-    if response.headers.has_key?(:content_length) && response.headers[:content_length].to_i < 0
+    if response.headers.key?(:content_length) && response.headers[:content_length].to_i.negative?
       raise CustomErrors::BadHeaderError
     end
   end
 
+  def sanitize_notes(json)
```

Comment:
if I'm being real picky, you're only receiving JSON in one of the three cases that are handled below... thus not a great name.